src/Maths/src/Box3.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+
+namespace Silk.NET.Maths
+{
+    /// <summary>
+    /// A structure representing a Box3 with a <see cref="Min"/> and a` <see cref="Max"/>
+    /// </summary>
+    public struct Box3<T>
+        : IEquatable<Box3<T>>
+        where T : unmanaged, IFormattable, IEquatable<T>, IComparable<T>
+    {
+        /// <summary>
+        /// The min.
+        /// </summary>
+        public Vector3<T> Min;
+        /// <summary>
+        /// The max.
+        /// </summary>
+        public Vector3<T> Max;
+        
+        /// <summary>
+        /// Constructs a Box3 from a min and a max
+        /// </summary>
+        /// <param name="min">The min of the rect.</param>
+        /// <param name="max">The max of the rect.</param>
+        public Box3(Vector3<T> min, Vector3<T> max)
+        {
+            Min = min;
+            Max = max;
+        }
+
+        /// <summary>
+        /// The center of this box.
+        /// </summary>
+        public Vector3<T> Center
+        {
+            get => Min + Max / Scalar<T>.Two;
```

Comment:
Can't be a property due to serializers getting thrown off by them.
Please convert to a method, the method version may share the same name for less typing.